javaparser-core/src/main/java/com/github/javaparser/ast/nodeTypes/NodeWithVariables.java

```diff
@@ -145,8 +146,8 @@ private Type toArrayLevel(Type type, int level) {
             // the pretty-printed string got for a node. We just check all them are the same and if they
             // are we just just is not null
             Object[] values = types.stream().map(v -> {
-                Type t = helper.toArrayLevel(v, currentLevel);
-                return t == null ? null : t.toString();
+                Optional<Type> t = helper.toArrayLevel(v, currentLevel);
+                return t.map(Node::toString).orElse(null);
```

Comment:
I'd say pretty printing is not something that should end up as input for any algorithm, but that's for another time.